web-components/src/components/datepicker/datepicker-calendar/DatePickerCalendar.ts

```diff
@@ -48,7 +48,15 @@ export namespace DatePickerCalendar {
 
     updated(changedProperties: PropertyValues) {
       super.updated(changedProperties);
-      if (changedProperties.has("datePickerProps") || changedProperties.has("viewAnchorDate")) {
+      if (changedProperties.has("datePickerProps") ) {
+        if(this.datePickerProps?.selected.invalidReason===null){
```

Comment:
only null check required or undefined check also required?
check this way works better if(!this.datePickerProps?.selected.invalidReason)   ?

Reply:
invalidReason: string | null; 
 
<img width="722" alt="Screenshot 2023-10-18 at 6 06 47 PM" src="https://github.com/momentum-design/momentum-ui/assets/148339647/0a0ef66c-f85f-4e2c-85fa-4a0aaae80c49">